ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ClientOptionsImpl.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.client.impl;
+
+import io.confluent.ksql.api.client.ClientOptions;
+import java.util.Objects;
+
+public class ClientOptionsImpl implements ClientOptions {
+
+  private String host = "localhost";
+  private int port = 8088;
+  private boolean useTls = false;
+  private boolean useClientAuth = false;
+  private boolean useBasicAuth = false;
+  private String trustStorePath = "";
```

Comment:
My original motivation for avoiding nulls was so the calling code wouldn't need perform null checks, but I've worked around this by updating the getter methods instead. I'm curious to know whether you think this is an improvement or not.

Reply:
I don't see what's wrong with the user dealing with nulls. Imho it's much more ugly for the user to have to deal with empty strings. If we're using empty strings to refer to absence of a value then what if empty string is a valid value? It gets nasty and ambiguous.

Reply:
It's true there's ambiguity but for values where the empty string isn't valid (e.g., default host, trust store path, etc.) the ambiguity doesn't really matter from the user's perspective -- null and empty string can be treated the same.

In terms of why it's more work for the user to deal with nulls, some libraries won't accept nulls since the user then has to null check (rather than having ClientOptionsImpl simply not return nulls). For example this line in `ClientImpl`
```
this.serverSocketAddress = SocketAddress
        .inetSocketAddress(clientOptions.getPort(), clientOptions.getHost());
```
fails with `java.lang.NullPointerException: no null host accepted` if `clientOptions.getHost()` is null. Should `ClientImpl` try to avoid this by using `clientOptions.getHost() == null ? "" : clientOptions.getHost()`, or should we pass in `clientOptions.getHost()` directly and let the failure occur if it's null?

Reply:
To clarify: I'm happy to switch to allowing nulls to be returned from ClientOptionsImpl. Just trying to under the full scope of your reasoning in an effort to learn :)

Reply:
Another reason why using empty strings to imply lack of property set is a bad idea - what if the string has some whitespace - a space, tab or CR? Do we count that as "null" too? This is super error prone and hard for the user to debug too (println on an empty string or a string with whitespace to the console or log looks the same)

Reply:
Regarding the SocketAddress example. This doesn't allow nulls as a socket address needs both a valid host and port to be well defined. The null check is there to make sure you're passing in both. If you use null to represent no host then it will fail nice and early. If you use empty string to represent no host, it will allow the object to be created, and won't fail until some time later in the program when it actually tries to use the address to connect.